cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	}
+
+	if len(varFilesResult) > 0 {
+		log.Entry().Infof("We will add the following string to the cf push call: '%s'", strings.Join(varFilesResult, " "))
+	}
+	return varFilesResult, nil
+}
+
+func quoteAndBashEscape(s string) string {
+	escapedSingleQuote := "'\"'\"'"
+	return strings.ReplaceAll(s, "'", escapedSingleQuote)
+}
+
+func checkAndUpdateDeployTypeForNotSupportedManifest(config *cloudFoundryDeployOptions) (string, error) {
+
+	manifestFile := config.Manifest
+	if len(manifestFile) == 0 {
+		manifestFile = "manifest.yml"
+	}
+	var manifestFileExists bool
+	var err error
+
+	if len(manifestFile) > 0 {
+		manifestFileExists, err = fileUtils.FileExists(manifestFile)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if config.DeployType == "blue-green" && manifestFileExists {
+
+		m, _ := _getManifest(manifestFile)
+
+		apps, err := m.GetApplications()
+
+		if err != nil {
+			return "", err
+		}
+		if len(apps) > 1 {
+			return "", fmt.Errorf("Your manifest contains more than one application. For blue green deployments your manifest file may contain only one application")
+		}
+
+		hasNoRouteProperty, err := m.ApplicationHasProperty(0, "no-route")
+		if err != nil {
+			return "", err
+		}
+		if len(apps) == 1 && hasNoRouteProperty {
+
+			const deployTypeStandard = "standard"
+			log.Entry().Warningf("Blue green deployment is not possible for application without route. Using deployment type '%s' instead.", deployTypeStandard)
+			return deployTypeStandard, nil
+		}
+	}
+
+	return config.DeployType, nil
+}
+
+func deployMta(config *cloudFoundryDeployOptions, mtarFilePath string, command command.ExecRunner) error {
+
+	deployCommand := "deploy"
+	deployParams := []string{}
+
+	if len(config.MtaDeployParameters) > 0 {
+		deployParams = append(deployParams, strings.Split(config.MtaDeployParameters, " ")...)
+	}
+
+	if config.DeployType == "bg-deploy" || config.DeployType == "blue-green" {
+
+		deployCommand = "bg-deploy"
+
+		const noConfirmFlag = "--no-confirm"
+		if !contains(deployParams, noConfirmFlag) {
+			deployParams = append(deployParams, noConfirmFlag)
+		}
+	}
+
+	cfDeployParams := []string{
+		deployCommand,
+		mtarFilePath,
+	}
+
+	if len(deployParams) > 0 {
+		cfDeployParams = append(cfDeployParams, deployParams...)
+	}
+
+	cfDeployParams = append(cfDeployParams, handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)...)
+
+	return cfDeploy(config, cfDeployParams, nil, nil, command)
+}
+
+func handleMtaExtensionDescriptors(mtaExtensionDescriptor string) []string {
+	var result = []string{}
+	for _, part := range splitAtWhitespace(strings.Trim(mtaExtensionDescriptor, " ")) {
+		if part == "-e" || part == "" {
+			continue
+		}
+		// REVISIT: maybe check if the extension descriptor exists
+		result = append(result, "-e", part)
+	}
+	return result
+}
+
+// would make sense to have that method in some kind of helper instead having it here
+func contains(collection []string, key string) bool {
```

Comment:
There is already piperutils.ContainsString(), I think.

Reply:
done